Client/Ecosia/UI/NTP/ClimateImpactCounter/SeedProgressManager/UserDefaultsSeedProgressManager.swift

```diff
@@ -0,0 +1,123 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+
+final class UserDefaultsSeedProgressManager: SeedProgressManagerProtocol {
+    
+    private static let className = String(describing: UserDefaultsSeedProgressManager.self)
+    static var progressUpdatedNotification: Notification.Name { .init("\(className).SeedProgressUpdated") }
+    static var levelUpNotification: Notification.Name { .init("\(className).SeedLevelUp") }
+    private static let numberOfSeedsAtStart = 1
+    
+    // UserDefaults keys
+    private static let totalSeedsCollectedKey = "TotalSeedsCollected"
+    private static let currentLevelKey = "CurrentLevel"
+    private static let lastAppOpenDateKey = "LastAppOpenDate"
+    
+    static var seedLevels: [SeedLevelConfig.SeedLevel] = SeedCounterNTPExperiment.seedLevelConfig?.levels.compactMap { $0 } ?? []
+    
+    private init() {}
+    
+    // MARK: - Static Methods
+
+    // Load the current level from UserDefaults
+    static func loadCurrentLevel() -> Int {
+        let currentLevel = UserDefaults.standard.integer(forKey: currentLevelKey)
+        return currentLevel == 0 ? 1 : currentLevel
+    }
+
+    // Load the total seeds collected from UserDefaults
+    static func loadTotalSeedsCollected() -> Int {
+        let seedsCollected = UserDefaults.standard.integer(forKey: totalSeedsCollectedKey)
+        return seedsCollected == 0 ? numberOfSeedsAtStart : seedsCollected
+    }
+
+    // Load the last app open date from UserDefaults
+    static func loadLastAppOpenDate() -> Date {
+        return UserDefaults.standard.object(forKey: lastAppOpenDateKey) as? Date ?? .now
+    }
+    
+    // Save the seed progress and level to UserDefaults
+    static func saveProgress(totalSeeds: Int, currentLevel: Int, lastAppOpenDate: Date) {
+        let defaults = UserDefaults.standard
+        defaults.set(totalSeeds, forKey: totalSeedsCollectedKey)
+        defaults.set(currentLevel, forKey: currentLevelKey)
+        defaults.set(lastAppOpenDate, forKey: lastAppOpenDateKey)
+        NotificationCenter.default.post(name: progressUpdatedNotification, object: nil)
+    }
+
+    // Helper method to get the seed threshold for the current level
+    private static func requiredSeedsForLevel(_ level: Int) -> Int {
+        if let seedLevel = seedLevels.first(where: { $0.level == level }) {
+            return seedLevel.requiredSeeds
+        }
+        return seedLevels.last?.requiredSeeds ?? 0  // If the level exceeds defined levels, use the last one
```

Comment:
It seems that we do not block the users level at the moment, isn't this part of this ticket's scope? From what I understand from the logic atm the user would keep levelling up but using the last item's threshold for every level we don't explicitly set in the payload.

I believe we actually want to have a disabled greyed out state when we reach the end of the last supported level but not sure if that is a separate ticket.

Reply:
> Two levels on the payload will make it not possible to level up more than the last defined.

If that's the case then seems good! From the logic won't it just continue levelling up using the required seeds for the last level of the payload instead of stopping?

I was only talking about [this greyed out state](https://www.figma.com/design/210Hc1efvYK3kIu0a4IdUo/Climate-impact-tracking-experiments?node-id=213-40488&t=UGBAloMSFb0jigxP-4) which already shows on the NTP, but if that's no longer part of the scope no problem.

Reply:
Great idea, thanks! ❤️